src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -249,26 +261,112 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
                                                 questionAttributes.courseId, numOfResponseBoxes,
                                                 existingResponse.getResponseDetails(), student);
 
-            responses.add(new FeedbackSubmissionEditResponse(responseIndx, true, recipientOptionsForQuestion,
-                                                                 submissionFormHtml, existingResponse.getId()));
+            commentGiverName = getCommentGiverNameFromEmail(existingResponse.giver, questionAttributes.giverType);
+            commentRecipientName = getCommentRecipientNameFromEmail(existingResponse.recipient,
+                    questionAttributes.recipientType);
+
+            Map<String, String> commentGiverEmailToNameTable = bundle.getRoster().getEmailToNameTableFromRoster();
+            if (questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+
+                FeedbackResponseCommentRow responseCommentRow = getResponseCommentRowForResponse(questionAttributes,
+                        existingResponse.getId(), commentGiverName, commentRecipientName, commentGiverEmailToNameTable);
+
+                FeedbackResponseCommentRow frcForAdding = buildFeedbackResponseCommentAddFormTemplate(
+                        questionAttributes, existingResponse.getId(), commentGiverName,
+                        commentRecipientName, sessionTimeZone, true);
+
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
+                        true, recipientOptionsForQuestion, submissionFormHtml,
+                        existingResponse.getId(), responseCommentRow, frcForAdding));
+
+                responseSubmittedRecipient.add(commentRecipientName);
+            } else {
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
```

Comment:
I prefer to use `setResponseCommentRow`, `setResponseCommentRowForAdding` as there are optional. Don't pass `null`.